autoload/minimap/vim.vim

```diff
@@ -386,9 +451,22 @@ function! s:highlight_line(winid, pos) abort
     call matchadd(g:minimap_highlight, '\%' . a:pos . 'l', 100, g:minimap_cursorline_matchid, { 'window': a:winid })
 endfunction
 
+function! s:set_range_color(winid, startpos, endpos, set_color, match_id) abort
+    call matchadd(a:set_color, '\%>' . a:startpos . 'l\%<' . a:endpos . 'l', 100, a:match_id, { 'window': a:winid })
+endfunction
 function! s:highlight_range(winid, startpos, endpos) abort
+    " Delete the old one before drawing
     silent! call matchdelete(g:minimap_cursorline_matchid, a:winid) " require vim 8.1.1084+ or neovim 0.5.0+
-    call matchadd(g:minimap_highlight, '\%>' . a:startpos . 'l\%<' . a:endpos . 'l', 100, g:minimap_cursorline_matchid, { 'window': a:winid })
+    call s:set_range_color(a:winid, a:startpos, a:endpos, g:minimap_highlight, g:minimap_cursorline_matchid)
+endfunction
+function! s:set_diffadd_range(winid, startpos, endpos) abort
+    call s:set_range_color(a:winid, a:startpos, a:endpos, g:minimap_diffadd_color, g:minimap_add_matchid)
+endfunction
+function! s:set_diffremove_range(winid, startpos, endpos) abort
+    call s:set_range_color(a:winid, a:startpos, a:endpos, g:minimap_diffremove_color, g:minimap_rem_matchid)
+endfunction
+function! s:set_diff_range(winid, startpos, endpos) abort
+    call s:set_range_color(a:winid, a:startpos, a:endpos, g:minimap_diff_color, g:minimap_diff_matchid)
```

Comment:
These functions aren't actually used. I used them for a bit, but then went with the dictionary solution above. I left them in because they seemed like nice convenience functions, but they can be removed with no repercussions if desired.